rfc/rfc006-distributed-registry.md

```diff
@@ -0,0 +1,344 @@
+
+#### Signer
+This document MUST be signed either by the vendor itself (in case of an update) or by another vendor (in case of a new
+vendor or loss of private key material).
+
+#### Duplicates
+Vendors SHOULD take care not to register a duplicate organization, but if this happens the vendor and all related
+objects (organizations, services, endpoints, etc.) MUST be processed as long as they're valid. As such there can be
+multiple vendor registrations with the same vendor ID. 
+
+#### Validation
+When processing a vendor registration or update it MUST be validated as follows:
+
+1. Assert all required fields are present.
+2. Assert all fields value formats are valid.
+3. For updates: assert only modifiable fields are updated.
+4. Validate **prfs**:
+   * Assert there is at least one proof.
+   * Assert proofs are valid at the time they're introduced.
+   * Assert all proofs authenticate the same vendor.
+5. Validate **certs**:
+   * Assert there is at least one certificate.
+   * Assert all certificates conform to the vendor CA certificate specification (see [RFC008](rfc008-certificate-structure.md)).
+   * Assert all certificates have the same MUST contain the same vendor ID.
+
+If any of these steps fail the registration or update SHALL NOT be processed.
+
+#### CA certificate private key loss
+In case the vendor loses its CA private key it MUST register a new vendor CA certificate. This can be done in two ways,
+depending on what was lost:
+
+1. If the vendor still has a valid signing certificate it COULD sign the update using that private key.
+2. Otherwise, when the signing certificate has expired or the private key is lost as well, the vendor MUST also provide
+new proof to authenticate ownership of the vendor. The new proof MUST be the first operation in the **patch** field.
+
+Example update:
+```json
+{
+  "patch": [
+    { "op": "add", "path": "/prfs", "value": "some new proof" },
+    { "op": "replace", "path": "/crts", "value": ["CA certificate as base64 ASN.1"] }
+  ]
+}
+``` 
+
+Vendors SHOULD use `replace` to atomically revoke the lost certificate and add the new one.
+
+#### CA certificate private key leakage
+In case the vendor leaks its CA private key (or it gets stolen), it MUST revoke the associated certificate as soon as possible.
+In the same update it MUST register new proof and CA certificate as in the key loss process.
+Example update:
+```json
+{
+  "patch": [
+    { "op": "add", "path": "/prfs", "value": "some new proof" },
+    { "op": "replace", "path": "/crts", "value": ["CA certificate as base64 ASN.1"] }
+  ]
+}
+``` 
+
+If a vendor removes a certificate it SHALL NOT be re-added
+TODO: What if the attacker re-adds the certificate after the vendor removed it, so it can still sign 
+TODO: What if the attacker keeps adding anti-dated (timestamp in history) changes, branching from an earlier document before the certificate was revoked?
+TODO: Internal revocation list SHOULD be build when a certificate is removed
```

Comment:
it'll be removed from the truststore in the OS ref impl. Maybe call it _appropriate measures against usage_?